controllers/async_controller.go

```diff
@@ -99,6 +100,31 @@ func (r *AsyncReconciler) Reconcile(req ctrl.Request, obj runtime.Object) (resul
 		return ctrl.Result{}, r.Update(ctx, obj)
 	}
 
+	// Ensure the resource is tagged with the operator's namespace.
+	objNamespace := annotations[namespaceAnnotation]
+	podNamespace := config.PodNamespace()
+	if objNamespace != podNamespace && objNamespace != "" {
+		// We don't want to get into a fight with another operator -
+		// so treat some other operator's annotation in a very similar
+		// way as the skip annotation above. This will do the right
+		// thing in the case of two operators trying to manage the
+		// same namespace. It makes moving objects between namespaces
+		// or changing which operator owns a namespace fiddlier (since
+		// you'd need to remove the annotation) but those operations
+		// are likely to be rare.
+		message := fmt.Sprintf("Operators in %q and %q are both configured to manage this resource", podNamespace, objNamespace)
+		r.Recorder.Event(obj, corev1.EventTypeWarning, "Overlap", message)
+		return ctrl.Result{}, r.Update(ctx, obj)
+	} else if objNamespace == "" {
+		// Set the annotation to this operator's namespace and go around again.
+		if annotations == nil {
+			annotations = make(map[string]string)
+		}
+		annotations[namespaceAnnotation] = podNamespace
+		res.SetAnnotations(annotations)
+		return ctrl.Result{}, r.Update(ctx, obj)
```

Comment:
I see what you mean, but I think in the situation where there are two operators watching one namespace it would be better to have one of them create the resource than both of them do it (maybe in different subscriptions so names wouldn't clash).